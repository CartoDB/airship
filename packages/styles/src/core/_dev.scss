@import 'tools/_mixins';
@import 'variables/_variables';
