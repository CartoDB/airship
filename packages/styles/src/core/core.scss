@import 'variables/_variables';

* {
  box-sizing: border-box;
  font-family: sans-serif;
}

:root {
  --primary-color: #{$primary-color};
}
