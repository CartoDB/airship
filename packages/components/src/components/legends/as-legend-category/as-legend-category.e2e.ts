import { E2EPage, newE2EPage } from '@stencil/core/testing';

describe('as-legend-category', async () => {
  describe('Rendering', async () => {
    let page: E2EPage;

    it('should render mixed types of legend entries', async () => {
      page = await newE2EPage({ html: `
        <as-legend-category>
        </as-legend-category>
      ` });

      const element = await page.find('as-legend-category');
      element.setProperty('data', [{
        color: '#FF0000',
        label: 'point label',
        strokeColor: '#00FF00',
        type: 'point',
        width: 16
      },
      {
        color: '#FF0000',
        label: 'line label',
        type: 'line'
      },
      {
        color: '#FF0000',
        label: 'polygon label',
        strokeColor: '#00FF00',
        type: 'line'
      }]);
      await page.waitForChanges();
      expect(element.outerHTML).toMatchSnapshot();
    });

    describe('should define a CSS variable', () => {
      it('with the max width of its children', async () => {
        page = await newE2EPage({ html: `
          <as-legend-category>
          </as-legend-category>
        ` });

        const element = await page.find('as-legend-category');
        element.setProperty('data', [{
          color: '#FF0000',
          label: 'point label',
          strokeColor: '#00FF00',
          type: 'point',
          width: 16
        },
        {
          color: '#FF0000',
          label: 'point label',
          strokeColor: '#00FF00',
          type: 'point',
          width: 32
        },
        {
          color: '#FF0000',
          label: 'point label',
          strokeColor: '#00FF00',
          type: 'point',
          width: 20
        }]);

        await page.waitForChanges();
        // CSS variables can't be accessed any other way right now
        expect(element.innerHTML).toContain('--as--basic--legend--figure-width: 32px');
      });

      it('but never below 16', async () => {
        page = await newE2EPage({ html: `
          <as-legend-category>
          </as-legend-category>
        ` });

        const element = await page.find('as-legend-category');
        element.setProperty('data', [{
          color: '#FF0000',
          label: 'point label',
          strokeColor: '#00FF00',
          type: 'point',
          width: 10
        }]);

        await page.waitForChanges();
        // CSS variables can't be accessed any other way right now
        expect(element.innerHTML).toContain('--as--basic--legend--figure-width: 16px');
      });
    });
  });
});
