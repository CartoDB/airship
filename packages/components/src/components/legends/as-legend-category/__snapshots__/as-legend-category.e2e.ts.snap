// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as-legend-category Rendering should render mixed types of legend entries 1`] = `"<as-legend-category class=\\"hydrated\\"><div class=\\"as-legend-category--wrapper\\" style=\\"--as--basic--legend--figure-width: 16px;\\"><div class=\\"as-legend-category--entry\\"><as-legend-category-point-entry class=\\"hydrated\\"><div class=\\"as-legend-category-point-entry--wrapper\\"><div class=\\"as-legend-category--figure--wrapper\\"><div class=\\"as-legend-category-point-entry--point as-legend-category-point-entry--symbol\\" style=\\"background-color: rgb(255, 0, 0); border: 1px solid rgb(0, 255, 0); height: 18px; width: 18px;\\"></div></div><span class=\\"as-legend-category--label\\">point label</span></div></as-legend-category-point-entry></div><div class=\\"as-legend-category--entry\\"><as-legend-category-line-entry class=\\"hydrated\\"><div class=\\"as-legend-category-line-entry--wrapper\\"><div class=\\"as-legend-category--figure--wrapper\\"><div class=\\"as-legend-category-line-entry--line\\" style=\\"border-top: 1.5px solid rgb(255, 0, 0);\\"></div></div><span class=\\"as-legend-category--label\\">line label</span></div></as-legend-category-line-entry></div><div class=\\"as-legend-category--entry\\"><as-legend-category-line-entry class=\\"hydrated\\"><div class=\\"as-legend-category-line-entry--wrapper\\"><div class=\\"as-legend-category--figure--wrapper\\"><div class=\\"as-legend-category-line-entry--line\\" style=\\"border-top: 1.5px solid rgb(255, 0, 0);\\"></div></div><span class=\\"as-legend-category--label\\">polygon label</span></div></as-legend-category-line-entry></div></div></as-legend-category>"`;
